internal/controlplane/handlers_providers.go

```diff
@@ -230,6 +230,45 @@ func (s *Server) DeleteProviderByID(
 	}, nil
 }
 
+// PatchProvider patches a provider by name from a specific project.
+func (s *Server) PatchProvider(
+	ctx context.Context,
+	req *minderv1.PatchProviderRequest,
+) (*minderv1.PatchProviderResponse, error) {
+	entityCtx := engine.EntityFromContext(ctx)
+	projectID := entityCtx.Project.ID
+	providerName := entityCtx.Provider.Name
+
+	if providerName == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "provider name is required")
+	}
+
+	err := s.providerManager.PatchProviderConfig(ctx, providerName, projectID, req.GetPatch().GetConfig().AsMap())
```

Comment:
Should this be getting a provider instance and then calling `PatchConfig` on the provider instance?  I'm okay with having this on providerManager, I just don't have a feel for when we create an instance and when we don't.

Reply:
yeah, me neither. Getting a provider instance and then patching it is pretty much what the implementation of PatchProviderConfig does now. 

@dmjb do you have an opinion since you created the `ProviderManager` interface?